### ISSUE TITLE
Refactor `calc_geos` into `import/process_geos` and `covariates_geos`

### ISSUE THREAD

**Issue by mitchellmanware** [Spatiotemporal-Exposures-and-Toxicology/NRTAPmodel#273]

**Comment by mitchellmanware** [comment]
Initial step for package spin off is to refactor the `calc_geos` and `calc_narr` functions into separate `process/import_`, `covariate_`, and `calc_` functions. `process_` will return a SpatRaster/SpatVector object that has been imported (from raw downloads), cleaned, and variable/spatially subsetted. `covariate_` will return data extracted at point locations as `data.frame`. Functionality of `calc_` is to be discussed.

**Comment by mitchellmanware** [comment]
### Progress (updated 1/22) (tracking on branch "mm-process-functions")

- [ ] `import_geos`
- [ ] `covariate_geos`
- [ ] `import_narr`
- [ ] `covariate_narr`

**Comment by mitchellmanware** [comment]
@sigmafelix 
Yes, my idea was basically what was already created with in the spacetime manipulation functions. Thank you for the reminder!

**Comment by mitchellmanware** [comment]
@sigmafelix @eva0marques 
The `convert_stobj_to_stdt` cannot be immediately applied in my case because the sites locations do not always contain time information. I can create a similar function to convert a objects to spatial only, or add a `time = TRUE` parameter to `convert_stobj_to_stdt` for converting objects without time. The latter would be quicker, but as the format is literally "spacetime data table", it may not make sense to include time-less functionality.

**Comment by eva0marques** [comment]
@mitchellmanware yes that is one point I need to reconsider in the future library dedicated to ST-data. Thanks for your feedback, it is true that if we want to merge stdt objects with with "only" spatial info it's gonna be necessary to be more flexible.

**Comment by mitchellmanware** [comment]
### Note (January 23, 2024)

Expanding covariate calculations to other pressure level data (for GEOS-CF; NARR to be seen) reveals missing data in the higher atmosphere covariates.

**Comment by mitchellmanware** [comment]
Simplified version on branch mm-process-functions called `sites_vector`. Function converts `data.frame` or `data.table` object into `SpatVector` based on "lon" and "lat" columns. Has check (and unit tests) for column names and other data types. Will expand to accept `sf` and `SpatVector` data types in future.

**Comment by eva0marques** [comment]
Great, thank you Mitchell! We will include that in the future separated package.
